src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -1355,7 +1355,9 @@ def compute_several_extensions(self, extensions, save=True, verbose=False, **job
 
             for extension_name, extension_params in extensions_with_pipeline.items():
                 extension_class = get_extension_class(extension_name)
-                assert self.has_recording(), f"Extension {extension_name} need the recording"
+                assert (
+                    self.has_recording() or self.has_temporary_recording()
+                ), f"Extension {extension_name} need the recording"
```

Comment:
```suggestion
                ), f"Extension {extension_name} requires the recording"
```
while we are fixing this. at minimum it needs an "s" added on. I prefer the word requires, but you could also change to needs :)

Reply:
done, plus some other docs fixes